src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Block.cs

```diff
@@ -42,6 +38,58 @@ public override BoundNode VisitBlock(BoundBlock node)
             return new BoundBlock(node.Syntax, synthesizedLocal == null ? node.Locals : node.Locals.Add(synthesizedLocal), node.LocalFunctions, builder.ToImmutableAndFree(), node.HasErrors);
         }
 
+
+        /// <summary>
+        /// Visit a partial list of statements that possibly contain using declarations
+        /// </summary>
+        /// <param name="statements">The list of statements to visit</param>
+        /// <param name="startIndex">The index of the <paramref name="statements"/> to begin visiting at</param>
+        /// <returns>An <see cref="ImmutableArray{T}"/> of <see cref="BoundStatement"/></returns>
+        public ImmutableArray<BoundStatement> VisitStatementSubList(ImmutableArray<BoundStatement> statements, int startIndex = 0)
+        {
+            var builder = ArrayBuilder<BoundStatement>.GetInstance();
+            for (int i = startIndex; i < statements.Length; i++)
+            {
+                BoundStatement statement = (BoundStatement)VisitPossibleUsingDeclaration(statements[i], statements, i, out var replacedUsingDeclarations);
+                if (statement != null)
+                {
+                    builder.Add(statement);
+                }
+
+                if (replacedUsingDeclarations)
+                {
+                    break;
+                }
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        /// <summary>
+        /// Visits a node that is possibly a <see cref="BoundUsingLocalDeclarations"/>
+        /// </summary>
+        /// <param name="node">The node to visit</param>
+        /// <param name="statements">A list of statements that follow this node</param>
+        /// <param name="startIndex">The startIndex of statements</param>
```

Comment:
>startIndex [](start = 41, length = 10)

Minor point: `startIndex` is a little confusing since it's not clear whether that represents the index of `node` or the index of the following statement. Perhaps removing the `node` parameter would make it more clear.